src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,194 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
```

Comment:
```suggestion
```

Reply:
Without validation, the tests will not pass.